src/query/interpreter.cpp

```diff
@@ -652,12 +652,25 @@ Callback HandleStreamQuery(StreamQuery *stream_query, const Parameters &paramete
       return callback;
     }
     case StreamQuery::Action::START_STREAM: {
-      callback.fn = [interpreter_context, stream_name = stream_query->stream_name_]() {
-        interpreter_context->streams.Start(stream_name);
-        return std::vector<std::vector<TypedValue>>{};
-      };
-      notifications->emplace_back(SeverityLevel::INFO, NotificationCode::START_STREAM,
-                                  fmt::format("Started stream {}.", stream_query->stream_name_));
+      const auto batch_limit = GetOptionalValue<int64_t>(stream_query->batch_limit_, evaluator);
+
+      if (!batch_limit.has_value()) {
+        callback.fn = [interpreter_context, stream_name = stream_query->stream_name_]() {
+          interpreter_context->streams.Start(stream_name);
+          return std::vector<std::vector<TypedValue>>{};
+        };
+        notifications->emplace_back(SeverityLevel::INFO, NotificationCode::START_STREAM,
+                                    fmt::format("Started stream {}.", stream_query->stream_name_));
+
+      } else {
+        callback.fn = [interpreter_context, stream_name = stream_query->stream_name_, batch_limit]() {
+          interpreter_context->streams.StartWithLimit(stream_name, batch_limit.value());
+          return std::vector<std::vector<TypedValue>>{};
+        };
+        notifications->emplace_back(
+            SeverityLevel::INFO, NotificationCode::START_STREAM,
+            fmt::format("Started stream {} with batch_limit {}.", stream_query->stream_name_, batch_limit.value()));
```

Comment:
I think this is ont necessary, maybe confusing. This notification will be delivered to the users when the query execution is done. They will see the results, they will know the stream has started. Therefore I think this notification might be a bit confusing.